### PR TITLE
SQL Server Migrations: Bypass type mapper when generating constants for DML

### PR DESCRIPTION
**Pull request by bricelam**

### CODE REVIEW

File: src/EFCore.Relational/Migrations/MigrationsSqlGenerator.cs [hunk 1]
```diff
@@ -1652,6 +1645,14 @@ protected virtual IUpdateSqlGenerator SqlGenerator
         protected virtual string ColumnList([NotNull] string[] columns)
             => string.Join(", ", columns.Select(Dependencies.SqlGenerationHelper.DelimitIdentifier));
 
+        /// <summary>
+        ///     Generates a Transact-SQL integer constant for the value.
+        /// </summary>
+        /// <param name="value"> The value. </param>
+        /// <returns> A string representing the integer constant. </returns>
+        protected virtual string IntegerConstant(long value)
```
Comment (smitpatel) [inline src/EFCore.Relational/Migrations/MigrationsSqlGenerator.cs]: Does this have to be protected? (not private). `GenerateConstant`?

File: src/EFCore.Relational/Migrations/MigrationsSqlGenerator.cs [hunk 2]
```diff
@@ -1147,18 +1143,15 @@ protected virtual IUpdateSqlGenerator SqlGenerator
...
                 .Append(" INCREMENT BY ")
-                .Append(intTypeMapping.GenerateSqlLiteral(increment));
+                .Append(IntegerConstant(increment));
 
             if (minimumValue != null)
             {
                 builder
                     .Append(" MINVALUE ")
-                    .Append(longTypeMapping.GenerateSqlLiteral(minimumValue));
+                    .Append(IntegerConstant(minimumValue));
```
Comment (maumar) [inline src/EFCore.Relational/Migrations/MigrationsSqlGenerator.cs]: this doesnt compile

**Comment by bricelam** [comment]
lol, this is all wrong--it was supposed to only be in the SQL Server code.

**Comment by bricelam** [comment]
Updated.

**Pull request merged**